docs/source/metrics.rst

```diff
@@ -1,137 +1,78 @@
+        'sgd',
+        'cross_entropy',
 
-Batch Metrics
--------------
+        batch_metrics=[Accuracy(), F1()],
+        # Can also use a string in this case:
+        # batch_metrics=['accuracy', 'f1'],
 
-Batch metrics are computed at each batch and averaged at the end of an epoch.
-The interface is the same as PyTorch loss function, that is ``metric(y_pred, y_true)``.
-They are passed to :class:`~poutyne.Model` and :meth:`ModelBundle.from_network() <poutyne.ModelBundle.from_network()>` using the ``batch_metrics`` argument.
+        epoch_metrics=[torchmetrics.AUROC(num_classes=10)],
+    )
+    model.fit_dataset(train_dataset, valid_dataset)
 
-In addition to the predefined batch metrics below, all PyTorch loss functions can be used by string name in the :class:`batch_metrics argument <poutyne.Model>` under their `functional <https://pytorch.org/docs/stable/nn.functional.html>`_ name.
-The key in :class:`callback logs<poutyne.Callback>` associated with each of them is the same as its name but without the ``_loss`` suffix. For example, the loss function :func:`~torch.nn.functional.mse_loss` can be passed as a batch metric with the name ``'mse_loss'`` or simply ``'mse'`` and the keys are going to be ``'mse'`` and ``'val_mse'`` for the training and validation MSE, respectively.
-Note that you can also pass the PyTorch loss functions as a loss function in :class:`~poutyne.Model` in the same way.
 
-Here is an example using batch metrics:
+Interface
+---------
 
-.. code-block:: python
+There are two interfaces available for metrics.
+The first interface is the same as PyTorch loss function, that is ``metric(y_pred, y_true)``.
+When using that interface, the metric is assumed to be decomposable and is average for the whole epoch.
+The batch size is infered with :func:`poutyne.get_batch_size()` using `y_pred` and `y_true` as values.
 
-  from poutyne import Model, Accuracy
+The second interface is defined by the :class:`~poutyne.Metric` class.
+As documented in the class, it provides methods for updating and computing  the metric.
+This interface is compatible with `TorchMetrics <https://torchmetrics.readthedocs.io/>`__, a library implementing a lot of known metrics in PyTorch.
+See the `TorchMetrics documentation <https://torchmetrics.readthedocs.io/en/latest/references/modules.html>`__ for available TorchMetrics metrics.
 
-  model = Model(
-      network,
-      'sgd',
-      'cross_entropy',
+Note that if one implements a metric intended to be used as both a batch and an epoch metric, the methods :meth:`Metric.forward()` and :meth:`Metric.update()` need both be implemented.
+To avoid having to implement both methods, one can `implement a TorchMetrics metric <https://torchmetrics.readthedocs.io/en/latest/pages/implement.html>`__ at the potential cost of higher computational load as described in the `TorchMetrics documentation <https://torchmetrics.readthedocs.io/en/latest/pages/implement.html#internal-implementation-details>`__.
```

Comment:
```suggestion
Note that if one implements a metric intended as both a batch and epoch metric, the methods :meth:`Metric.forward()` and :meth:`Metric.update()` need to be implemented.
To avoid implementing both methods, one can `implement a TorchMetrics metric <https://torchmetrics.readthedocs.io/en/latest/pages/implement.html>`__ at the potential cost of higher computational load as described in the `TorchMetrics documentation <https://torchmetrics.readthedocs.io/en/latest/pages/implement.html#internal-implementation-details>`__.
```